lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TemporaryCacheKeys, Constants } from "../utils/Constants";
+import { ICacheStorage } from "./ICacheStorage";
+import { Account } from "../auth/Account";
+import { Authority } from "../auth/authority/Authority";
+import { CodeRequestParameters } from "../server/CodeRequestParameters";
+
+export class CacheHelpers {
+
+    private cacheStorage: ICacheStorage;
+
+    constructor(cacheImpl: ICacheStorage) {
+        this.cacheStorage = cacheImpl;
+    }
+
+    /**
+     * Create acquireTokenAccountKey to cache account object
+     * @param accountId
+     * @param state
+     */
+    generateAcquireTokenAccountKey(accountId: any, state: string): string {
+        return `${TemporaryCacheKeys.ACQUIRE_TOKEN_ACCOUNT}${Constants.RESOURCE_DELIM}${accountId}${Constants.RESOURCE_DELIM}${state}`;
+    }
+
+    /**
+     * Create authorityKey to cache authority
+     * @param state
+     */
+    generateAuthorityKey(state: string): string {
+        return `${TemporaryCacheKeys.AUTHORITY}${Constants.RESOURCE_DELIM}${state}`;
+    }
+
+    /**
+     * @hidden
+     * @ignore
+     *
+     * Sets the cachekeys for and stores the account information in cache
+     * @param account
+     * @param state
+     * @hidden
+     */
+    setAccountCache(account: Account, state: string) {
```

Comment:
another question: what's the use case here? is this idea of an `AccountCache` supposed to be how we're using a single cache but separating different items in the cache for different requests/apps/tabs/etc? hopefully that thought makes sense

Reply:
I will rename this, but this is just to cache the account object in our current cache. We only use a single cache today, and we currently only support single account scenarios. Multiple account scenarios may call for the design you are describing, but I don't think we have a finalized solution for it yet.

Reply:
okay--makes sense. I remembered talk about multiple account scenarios, and that's why i asked.